android/app/src/main/java/com/textile/textilenode/TextileNode.java

```diff
@@ -241,12 +246,11 @@ public void getPhotoBlocks (String offset, Integer limit, String threadName, Pro
             String jsonString = node.getPhotoBlocks(offset, limit, threadName);
             // convert response to json
             JSONObject obj = new JSONObject(jsonString);
-            WritableArray array = new WritableNativeArray();
-            JSONArray jsonArray = obj.getJSONArray("items");
-            for (int i = 0; i < jsonArray.length(); i++) {
-                Object value = jsonArray.get(i);
-                array.pushString((String) value);
+            JSONArray jsonArray = obj.optJSONArray("items");
+            if (jsonArray == null) {
+                jsonArray = new JSONArray();
             }
+            WritableArray array = TextileNode.convertJsonToArray(jsonArray);
             promise.resolve(array);
```

Comment:
Previously, we were sending an array of `String`s here. Needed to make it consistent with the iOS version which sends an array of objects that will be properly converted to JS objects by RN. In this case, that object type is `WritableArray` (which is an array of `WritableMap`s). RN automatically converts them to JS objects.